build-aux/m4/README.md

```diff
@@ -0,0 +1,8 @@
+## Autoconf archive
+
+All files in this folder are from the [Autoconf Archive](https://www.gnu.org/software/autoconf-archive/)
+([Github](https://github.com/autoconf-archive/autoconf-archive/tree/master/m4) may be ahead of the site).
+They are unmodified, but may not be the most recent serial.
+
+Future modifications to these files should be listed here and at the top of each file,
+to prevent accidentally overriding them.
```

Comment:
We shouldn't need to list modifications to the `.m4` files in 2 separate places. Any modifications should already be documented by the commits to this directory that introduce them.

Reply:
We did that before and (at least I) managed to accidentally forget about those changes in a followup. That said, I don't know if this note in the README can prevent that either. A linter script might.